internal/controller/managedcluster_controller.go

```diff
@@ -337,6 +346,48 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, managedCluster *h
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagedClusterReconciler) needToRequeue(ctx context.Context, managedCluster *hmc.ManagedCluster) (bool, error) {
+	childObjects := []objectToCheck{
+		{
+			gvr: schema.GroupVersionResource{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "clusters",
+			},
+			requeueIfNoObjects: true,
+			statuses: map[string]string{
+				"ControlPlaneInitialized": "Control Plane",
+			},
+		},
+		{
+			gvr: schema.GroupVersionResource{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "machinedeployments",
+			},
+			requeueIfNoObjects: false,
+			statuses: map[string]string{
+				"Available": "Machine Deployment",
+			},
+		},
+	}
+
+	var requeue bool
+	var errs error
+	for _, obj := range childObjects {
+		r, err := r.setStatusFromChildObjects(ctx, managedCluster, obj)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
```

Comment:
I think there is no need for a new direct dependency and this lines may also be simplified to just
```go
errs = errors.Join(errs, err)
```